src/AWS.Deploy.ServerMode.Client/ServerModeSession.cs

```diff
@@ -0,0 +1,122 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.\r
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using AWS.Deploy.ServerMode.Client.Exceptions;
+
+namespace AWS.Deploy.ServerMode.Client
+{
+    /// <summary>
+    /// Helper class that allows launching deployment tool in server mode.
+    /// </summary>
+    public class ServerModeSession
+    {
+        private const int TCP_PORT_ERROR = -100;
+
+        private readonly int _startPort;
+        private readonly int _endPort;
+
+        public ServerModeSession(int startPort = 5000, int endPort = 5100)
+        {
+            _startPort = startPort;
+            _endPort = endPort;
+        }
+
+        /// <summary>
+        /// Start deployment tool in server mode.
+        /// </summary>
+        /// <returns>Base URL of the  deployment tool local server.</returns>
+        /// <exception cref="InternalServerModeException">Throws when deployment tool server failed to start for un unknown reason.</exception>
+        /// <exception cref="PortUnavailableException">Throws when deployment tool server failed to start due to unavailability of free ports.</exception>
+        public async Task<string> Start()
+        {
+            var currentProcessId = Process.GetCurrentProcess().Id;
+
+            for (var port = _startPort; port <= _endPort; port++)
+            {
+                var baseUrl = $"http://localhost:{port}";
+
+                var command = $"dotnet aws server-mode --port {port} --parent-pid {currentProcessId}";
```

Comment:
We need to set the `--encryption-keyinfo-stdin` switch and then generate the encryption key info and pass that into stdin. This test might help you see what needs to be done. https://github.com/aws/aws-dotnet-deploy/blob/main/test/AWS.Deploy.CLI.IntegrationTests/ServerModeTests.cs#L108

We will also need to make sure we help create the RestClient with the key information as well.

Reply:
Fixed.